### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by K0ppai**
Dear reviewers,
We've worked on the following in this pull request,
### n+1
- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.
#### Before 
before
#### After
after

### Integration specs
- Use Capybara to write integration tests for each view in the project.

### CODE REVIEW

File: spec/views/users/index.html.erb_spec.rb [hunk 1]
```diff
@@ -1,5 +1,26 @@
...
+    expect(page).to have_content(@user.name)
+  end
+
+  scenario 'I can see the profile picture' do
+    expect(page).to have_css("img[src*='https://picsum.photos/id/23/200']")
+  end
+
+  scenario 'I can see the number of posts' do
+    expect(page).to have_content("Number of posts: #{@user.post_counter}")
+  end
```
Comment (codepantha) [inline spec/views/users/index.html.erb_spec.rb]: - [ ] Well done for your progress so far in the test cases. However, kindly note that the 3 scenarios highlighted in your code above should check the following: 
1. You can see the username of ALL OTHER USERS.
2. You can see the profile picture for EACH USER
3. You can see the number of posts EACH USER has written..

image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @K0ppai 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ Wrote integration specs for views 📌 
✔️ All tests are green 🟢 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by addod19 — APPROVED** [review]
## Hi @K0ppai and @yemidada, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [ ] Great job adding the required test cases highlighted in the previous review :rocket: 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by K0ppai** [comment]
Thanks for the reviews @addod19 and @codepantha. 😁

**Pull request closed without merging**